hw-model/src/lib.rs

```diff
@@ -554,6 +554,37 @@ pub trait HwModel {
         }
     }
 
+    fn step_until_fatal_error(&mut self, expected_error: u32) {
+        // Since the boot takes less than 30M cycles, we know something is wrong if
```

Comment:
it takes ~28M cycles now, but I`m not sure we can rely on this number for long term. As we saw in the last few weeks, this number can vary, and this can potentially hide a bug.
Can this logic go outside of the model, and probably on the test case, where we can pass the MAX_WAIT_CYCLE as a parameter ?

Reply:
Updated.